"Featured Product" Block: Add link to the product to the block

ryelle opened the pull request:
See #294 – This adds the link button to the product block (a link styled as a button). It uses the RichText component in the editor, so users can customize the link text. On the front end, it's a plain link with the user's custom text (if applicable). There's also a screen reader label reading "view product {name}", to avoid repeated/unclear link text.

The styling is pretty basic, using the same class as the core button - hopefully this way we can be consistent with themes that customize the button class.

### Screenshots

In the editor:

![screen shot 2019-01-11 at 7 06 05 pm](https://user-images.githubusercontent.com/541093/51065706-4d8f5580-15d4-11e9-92e0-97bdfa718dc2.png)

On the front end:

![screen shot 2019-01-11 at 7 05 35 pm](https://user-images.githubusercontent.com/541093/51065714-52eca000-15d4-11e9-863b-65976e932d9b.png)

### How to test the changes in this Pull Request:

1. Add a Featured Product block, select a product
2. Expect: The product preview should have a link with default text "Shop now"
3. Edit the link text
4. Save your post, view it
5. Expect: The product block should have a link using the text you entered.

ryelle committed:
Featured Product Block: Add link to the product to the block

ryelle committed:
Add a better link label for screen reader users

timmyc reviewed (commented):
Code is looking good, and I can modify the button text but the button seems to be missing some styling for me locally - note the missing padding/border radius:

__Front End__
![image](https://user-images.githubusercontent.com/22080/51066452-0cd91180-15bf-11e9-9eb4-1e819ba60d93.png)

__Editor__
![image](https://user-images.githubusercontent.com/22080/51066477-2ed29400-15bf-11e9-95dc-f0c600862945.png)

ryelle commented:
@timmyc Interesting– is that Twenty Nineteen? did you past that content in from a text editor or something that might've had styles attached?

timmyc commented:
Correct, 2019 is the theme - here is a screen grab of adding a new featured product block - so even with the default text, the button is still square - and inspecting the css for the element i'm not seeing a border radius in any of the classes applied to the element:

![featured-product](https://user-images.githubusercontent.com/22080/51066634-39d9f400-15c0-11e9-83ad-63e4db8cd703.gif)

![image](https://user-images.githubusercontent.com/22080/51066651-4b230080-15c0-11e9-9133-c7ac19b2445f.png)

timmyc commented:
😊 updates Gutenberg to the latest, and all is happy now. Sorry for the fire drill @ryelle

timmyc reviewed (approved):
I apparently can't keep all my test sites updated properly - all is testing well for me now, thanks for helping me debug Kelly.

LevinMedia reviewed (commented):
I'm seeing a difference in styles between the feature product block button, and the buttons in the other blocks (for example, category block - I'm using storefront) I would expect those to match. 

Also when aligning the text left or right, I'd expect the button to re-align itself as well. 

<img width="1122" alt="screen shot 2019-01-11 at 10 12 24 pm" src="https://user-images.githubusercontent.com/4500952/51069974-46754100-15ee-11e9-8f31-6e379b2f6412.png">

Outside of the scope of this PR but noting here... in the editor view, the buttons in the category block have rounded corners, (see screen shot above) but on the front end they're square. (Screenshot below) There are also some very subtle differences in font weight and color. 

<img width="1159" alt="screen shot 2019-01-11 at 10 12 45 pm" src="https://user-images.githubusercontent.com/4500952/51069983-8ccaa000-15ee-11e9-866f-065cc48d9025.png">

ryelle commented:
> in the editor view, the buttons in the category block have rounded corners, (see screen shot above) but on the front end they're square. (Screenshot below) There are also some very subtle differences in font weight and color.

Those should be left up to the theme to control, since the difference comes from how themes style the shortcode – so I would say Storefront should add editor styles that make the product grid blocks match the frontend.

ryelle committed:
Match core button styles, fix alignment of button-link

ryelle commented:
I've fixed the button display when compared to core buttons (the issue with matching the product grid buttons is the same as above), and fixed the button alignment.

LevinMedia commented:
Looks good! Thanks!
